src/components/Canvas/CanvasTemplate.vue

```diff
@@ -121,83 +149,96 @@ export default {
+      // enable new related images and center the screen.
       clearTimeout(this.focusMoveTimeout);
       this.focusMoveTimeout = undefined;
       // Retrieve current center window position in the page
       const currentCenterLeftPosition =
         -this.$refs["page"].getBoundingClientRect().x + this.windowWidth / 2;
       const currentCenterTopPosition =
         -this.$refs["page"].getBoundingClientRect().y + this.windowHeight / 2;
-      // retrieve the center image positions in the page
-      const currentImageLeftPositionString = this.$refs["image-element"]
-        .position.left;
-      const currentCenterImageLeftPosition = +currentImageLeftPositionString.substring(
-        0,
-        currentImageLeftPositionString.length - 2
-      );
-      const currentImageTopPositionString = this.$refs["image-element"].position
-        .top;
-      const currentCenterImageTopPosition = +currentImageTopPositionString.substring(
-        0,
-        currentImageTopPositionString.length - 2
-      );
-      // Check collision
-      this.collisionAnalysis(
+
+      // Initial central image analyzis
+      this.imageCollisionAnalyzis(
         currentCenterLeftPosition,
         currentCenterTopPosition,
-        currentCenterImageLeftPosition,
-        currentCenterImageTopPosition,
-        this.$refs["image-element"]
+        this.$refs,
+        "image-element",
+        this.initialImageFocus,
+        0
       );
+
       // Analysis of each related images
-      for (let i = 0; i <= 2; i++) {
-        // Retrieve current image position in the page
-        const currentImageLeftPositionString = this.$refs["image-block"].$refs[
-          "image-element-" + i
-        ].position.left;
-        const currentImageLeftPosition = +currentImageLeftPositionString.substring(
+      // first loop to analyze each blocks
+      // Second loop to analyze each images in the block
+      for (let i = 0; i <= this.imageBlockController.length; i++) {
+        const currentImageBlock = this.$refs["image-block-" + i];
+        if (currentImageBlock) {
+          const nbElementsInBlock = Object.keys(currentImageBlock.$refs).length;
+          for (let j = 0; j < nbElementsInBlock; j++) {
+            this.imageCollisionAnalyzis(
+              currentCenterLeftPosition,
+              currentCenterTopPosition,
+              currentImageBlock.$refs,
+              "image-element-" + j,
+              this.imageBlockController[i].relatedImages[j],
+              this.imageBlockController[i].imagePositions[j]
+            );
+          }
+        }
+      }
+
+      // Manage size of the focus rectangle
+      this.$refs["rectangle"].hasFocus = this.currentFocus.includes(true);
+    },
+    imageCollisionAnalyzis(
+      currentCenterLeftPosition,
+      currentCenterTopPosition,
+      refToAnalyze,
+      ref,
+      focusElement,
+      imagePosition
+    ) {
+      const imageToAnalyze = refToAnalyze[ref];
```

Comment:
directly give image to analyze as parameter
probably enough to have:
`imageCollisionAnalyzis(
      imageDivToAnalyze,
      relatedImage
    )`

Reply:
Yeah, pass directly the div to analyse works well (I test to do that during the implementation but now I find the bug related to this approach so that works).

But the two others parameters is related to the imageBlock object and not the div. For this reason, the two are necessary.